sdk/attestation/azure-security-attestation/test/samples/attestation/attestation.cpp

```diff
@@ -54,12 +54,15 @@ int main()
   }
   catch (Azure::Core::RequestFailedException const& e)
   {
-    std::cout << "Attestation Client Exception happened:" << std::endl << e.Message << std::endl;
-    return 1;
+    std::cout << "Request Failed Exception happened:" << std::endl << e.what() << std::endl;
+    if (e.StatusCode != Azure::Core::Http::HttpStatusCode::None)
```

Comment:
nit

Here's another alternative for this condition:

```cpp
if(e.RawResponse) {
      std::cout << "Error Code: " << e.ErrorCode << std::endl;
      std::cout << "Error Message: " << e.Message << std::endl;
}
```

Might be a little less verbose

Reply:
In order to set any field related to an http response, the RFE must be constructed with a RawResponse.
When constructing a REF with a RawResponse, there will be a status code, ErrorCode, Message, etc.  When RFE is constructed with a simple string, the RawResponse will be `null_ptr` and RFE will be only able to provide `wath()` to print that string used to initialize it.

That's why, before acceding any other thing than `what()`, we can either check the status code, or the RawResponse

Reply:
Where is this relationship documented, or these invariants checked?

How does a developer know that they can distinguish between a RFE thrown by a response containing an error and an RFE generated because a transport error has occurred?

I've looked at the documentation for RFE and none of what you just wrote is expressed anywhere I can see. For instance, the documentation for `RawResponse` is simply "The entire HTTP raw response.". How do I know that RawResponse is set iff this is an HTTP response error rather than some other kind of HTTP error?

Also: Why is checking for `RawResponse` preferred over checking the status code?

Reply:
Next iteration of this PR will have a change to RFE to add some words to this effect.